internal/test/e2e/basic_test.go

```diff
@@ -564,6 +564,89 @@ func TestSNISync(t *testing.T) {
 	})
 }
 
+func TestVaultSync(t *testing.T) {
+	// ensure that Vaults can be synced to Kong gateway
+	cleanup := run.Koko(t)
+
+	defer cleanup()
+
+	vault := &v1.Vault{
+		Id:     uuid.NewString(),
+		Name:   "env",
+		Prefix: "test-vault-1",
```

Comment:
Please add a `Config` here.
Right now the sync to DPs won't work and this test doesn't catch it.

Reply:
The error looks like this:

```
Error: rr: 2022/10/20 08:31:22 [error] 1109#0: *6 [lua] wrpc_data_plane.lua:180: [wrpc-clustering] unable to update running config: bad config received from control plane in 'vaults':
dp-stderr:   - in entry 1 of 'vaults':
dp-stderr:     in 'config':
dp-stderr:       in 'env': unknown field, context: ngx.timer
```

This is happening because we are modelling the Vault object differently than what the DP expects.